app/models/ability.rb

```diff
@@ -34,11 +34,15 @@ def all_user_abilities
   def user_with_roles_abilities
     return if user.roles.empty?
 
-    can :read, ActiveStorage::Attachment
-    can :read, MarcRecord
-    can :read, Stream
-    can :read, Upload
     can :read, :pages_data
+    organization_ability_restrictions
+  end
+
+  def organization_ability_restrictions(restrictions = { restrict_downloads: false })
+    can :read, ActiveStorage::Attachment, { record: { organization: restrictions } }
+    can :read, MarcRecord, upload: { organization: restrictions }
+    can :read, Stream, organization: restrictions
+    can :read, Upload, organization: restrictions
```

Comment:
It took me a moment to figure out what was going on with the `restrictions` argument here. I guess I'm hoping that being this clever pays off in the next step (when we're also allowing members of organizations that have allowed access etc..), but maybe it'd be clearer if we duplicated the permissions (in different methods, e.g. `def allow_read_access_to_unrestricted_organization_data`, `def allow_read_access_to_organization_data_by_organization_ids(ids)`).

But I'm happy to just be assured it'll all make total sense in a bit 🤷

Reply:
I may get to the next step and find I agree with you 😸